Add config intro page

ComfortablyCoding opened the pull request:
This PR add the missing `Configuration Files`, `Environment Variable Files`, `Type Casting and Nesting` and `Environment Syntax Prefix` sections from the old docs. These sections explain what options users have for config files and formats as well as documents our auto casting functionality. 

It was added as an intro section to the configuration page as that looked to be the most appropriate location.

Changes were copied directly [from the original](https://github.com/directus/legacy-docs/blob/main/self-hosted/config-options.md) with the exception of cef2f382b60cf293d063712621ca23096e98ec2b, to clarify usage due to https://github.com/directus/directus/issues/25856

---
Fixes: #412

ComfortablyCoding committed:
add env config options

ComfortablyCoding committed:
clarfiy that left hand side env must be a directus key for `_FILE`

licitdev commented on content/configuration/0.intro.md:
```diff
@@ -0,0 +1,226 @@
+---
+title: Configuration Options
```
<img width="341" height="229" alt="image" src="https://github.com/user-attachments/assets/03d10aae-3105-4c2a-8aa6-5ebf6b6ad652" />

```suggestion
title: Introduction
```

licitdev reviewed (commented):
Thanks for resurrecting this! 💯 

Should this page be the landing page for the nav? 
https://github.com/directus/docs/blob/5549574887c37e0a65aee22821422dd2573363fc/app/app.config.ts#L112

Other references to `/configuration/general` look to refer to specific envs in the general page.

ComfortablyCoding committed:
Update content/configuration/0.intro.md

Co-authored-by: ian <licitdev@gmail.com>

ComfortablyCoding committed:
add as landing page for config

ComfortablyCoding commented:
> Thanks for resurrecting this! 💯
> 
> Should this page be the landing page for the nav?
> 
> https://github.com/directus/docs/blob/5549574887c37e0a65aee22821422dd2573363fc/app/app.config.ts#L112
> 
> Other references to `/configuration/general` look to refer to specific envs in the general page.

Yes I believe it should. Added in a8d2b053fe102f94b7675dae5457e1294f35bcdb, updated the homepage as well via 1e4681a85193f4144392314bdd0aa49c16daa68f

ComfortablyCoding committed:
add to homepage as well

ComfortablyCoding committed:
fix code group

licitdev reviewed (approved):
🚀 

<img width="233" height="505" alt="image" src="https://github.com/user-attachments/assets/cdc3c9a8-9ba7-47e0-acc1-2ab915b4a7d3" />

ComfortablyCoding commented on content/configuration/0.intro.md:
```diff
@@ -0,0 +1,222 @@
+
+## Environment Variable Files
+
+Any of the environment variable values can be imported from a file, by appending `_FILE` to the directus environment variable
```
Thoughts on adding `documented`? 

```suggestion
Any of the environment variable values can be imported from a file, by appending `_FILE` to a documented Directus environment variable (e.g. `DB_PASSWORD`).
```

ComfortablyCoding replied:
We can alternatively point to our [directus variable list in our code](https://github.com/directus/directus/blob/main/packages/env/src/constants/directus-variables.ts) which is what is referenced when checking what is accepted.

ComfortablyCoding replied:
```suggestion
Any of the environment variable values can be imported from a file, by appending `_FILE` to a [Directus environment variable](https://github.com/directus/directus/blob/main/packages/env/src/constants/directus-variables.ts).
```

ComfortablyCoding replied:
Resolving as referenced incorrect lines

ComfortablyCoding replied:
Added via 50276f37ab23b9ed858905d9f7fbe615528b8876

ComfortablyCoding committed:
add direct link to accepted vars for `_FILE`

Co-Authored-By: ian <26413686+licitdev@users.noreply.github.com>
Co-Authored-By: Brainslug <br41nslug@users.noreply.github.com>
